c-scape/src/math/mod.rs

```diff
@@ -728,6 +728,57 @@ unsafe extern "C" fn nearbyintf(x: f32) -> f32 {
     libm::rintf(x)
 }
 
+#[no_mangle]
+unsafe extern "C" fn finite(x: f64) -> i32 {
+    x.is_finite() as i32
+}
+#[no_mangle]
+unsafe extern "C" fn finitef(x: f32) -> i32 {
+    x.is_finite() as i32
+}
+
+#[no_mangle]
+unsafe extern "C" fn isnan(x: f64) -> i32 {
+    x.is_nan() as i32
+}
+
+#[no_mangle]
+unsafe extern "C" fn isnanf(x: f32) -> i32 {
+    x.is_nan() as i32
+}
+
+#[no_mangle]
+unsafe extern "C" fn isinf(x: f64) -> i32 {
+    if x == f64::INFINITY {
+        1
+    } else if x == f64::NEG_INFINITY {
+        -1
+    } else {
+        0
+    }
+}
+
+#[no_mangle]
+unsafe extern "C" fn isinff(x: f32) -> i32 {
+    if x == f32::INFINITY {
+        1
+    } else if x == f32::NEG_INFINITY {
+        -1
+    } else {
+        0
+    }
+}
+
+#[no_mangle]
+unsafe extern "C" fn lgamma(x: f64) -> f64 {
+    libm::lgamma(x)
+}
+
+#[no_mangle]
+unsafe extern "C" fn lgammaf(x: f32) -> f32 {
+    libm::lgammaf(x)
+}
```

Comment:
The libc `lgamma` is supposed to set a global variable `signgam` with the sign value. It doesn't look like the libm crate's `lgamma` does that. But you could implement `lgamma` in terms of `lgamma_r` which returns the sign value as a separate return value, and set `signgam` yourself.

Reply:
added signalling flag signgam. as https://github.com/simonbyrne/apple-libm/blob/master/Source/Intel/math.h doesn't seem to provide a default, using 0 (which means isn't meaningful before lgamma has been called, similar to c)

Reply:
Looks good. Just one nit: `signgam` doesn't need to be `pub`. The `#[no_mangle]` suffices to make things globally visible in the C global namespace.